pkg/controller/cluster/main-controller.go

```diff
@@ -490,6 +490,35 @@ func (mk *minioKeychain) Resolve(_ authn.Resource) (authn.Authenticator, error)
 	}), nil
 }
 
+// getKeychainForTenant attempts to build a new authn.Keychain from the image pull secret on the Tenant
+func (c *Controller) getKeychainForTenant(ref name.Reference, tenant *miniov1.Tenant) (authn.Keychain, error) {
+	// Get the secret
+	secret, err := c.kubeClientSet.CoreV1().Secrets(tenant.Namespace).Get(context.Background(), tenant.Spec.ImagePullSecret.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.New("can't retrieve the tenant image pull secret")
```

Comment:
why not return from this function here with default key chain

```suggestion
		return authn.DefaultKeychain, errors.New("can't retrieve the tenant image pull secret")
```